front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -130,15 +195,47 @@ class IntentMenu extends Component {
                                 }}
                             />
                         </div>
-                        <ul className="text-center font-bold text-xl mt-10"> 
+                        <div>
+                            <div
+                                className="inline-block align-middle text-xl mx-10 my-5"
+                            >
+                                <input
+                                    data-testid="star-filter"
+                                    id="star-filter"
+                                    type="checkbox"
+                                    onChange={this.handleStarChange}
+                                />
+                                <label
+                                    className="mx-2"
+                                >
+                                    Star
+                                </label>
+                            </div>
+                            <div
+                                className="inline-block align-middle text-xl mx-10 my-5"
+                            >
+                                <input
+                                    data-testid="flag-filter"
+                                    id="flag-filter"
+                                    type="checkbox"
+                                    onChange={this.handleFlagChange}
+                                />
+                                <label
+                                    className="mx-2"
+                                >
+                                    Flag
+                                </label>
+                            </div>
```

Comment:
I think making these visual buttons (like the star/flag) would be more effective. I also think adding a tooltip might be nice.

Also from an a11y perspective, pressing enter on the checkbox does not check it (which we should fix).

Reply:
There is also an issue with exiting the intent menu when on a filtered version, because when you re-open the menu, the questions are filtered, but the inputs are reverted to being blank, so whenever you try toggling them again, they now do the opposite function. (a.k.a checking off star will reveal all star and non-star questions rather than just starred questions)

Reply:
This can be most easily fixed by moving a lot of functionality into redux :+1: